drivers/modem/hl7800.c

```diff
@@ -1829,17 +1829,40 @@ static void dns_work_cb(struct k_work *work)
 #if defined(CONFIG_DNS_RESOLVER) && !defined(CONFIG_DNS_SERVER_IP_ADDRESSES)
 	int ret;
 	struct dns_resolve_context *dnsCtx;
-	static const char *const dns_servers_str[] = { ictx.dns_v4_string,
+	struct sockaddr temp_addr;
+	bool valid_address = false;
+	static const char *const dns_servers_str[] = {
 #ifdef CONFIG_NET_IPV6
-						       ictx.dns_v6_string,
+		ictx.dns_v6_string,
 #endif
-						       NULL };
+#ifdef CONFIG_NET_IPV4
+		ictx.dns_v4_string,
+#endif
+		NULL};
+
+#ifdef CONFIG_NET_IPV6
+	if (IS_ENABLED(CONFIG_NET_IPV6)) {
```

Comment:
Why double check, the other one can be removed, preferably the `#ifdef` one

Reply:
Need to leave the #ifdef one because `dns_v6_string` is not available if no `CONFIG_NET_IPV6`. I will clean this up.